prusti-viper/src/utils/type_visitor.rs

```diff
@@ -268,7 +268,10 @@ pub fn walk_closure<'tcx, E, V: TypeVisitor<'tcx, Error = E>>(
 ) -> Result<(), E> {
     let cl_substs = substs.as_closure();
     // TODO: when are there bound typevars? can type visitor deal with generics?
-    let fn_sig = cl_substs.sig().no_bound_vars().unwrap();
+    let fn_sig = match cl_substs.sig().no_bound_vars() {
+        Some(t) => t,
+        None => unimplemented!("Bound variables not supported at {:?}", def_id)
+    };
```

Comment:
To avoid the `Some(t) => t`, you could use `no_bound_vars().unwrap_or_else(|| unimplemented!("<message>", ...))` or even `no_bound_vars().expect(&format!("<message>", ...))`.